### PR TITLE
[types] - refactor: enforce non-empty strings for certain user contex…

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

Updated the `timezone` and `username` fields in the user message context to use `NonEmptyString` type for better data validation

Using these new types ensure that the following POST request:
```
curl  -vvv  -XPOST http://localhost:3000/api/v1/w/[wId]/assistant/conversations/[cId]/messages -H "Authorization: Bearer $DUST_API_KEY" -H "Content-Type: application/json" \
 -d '{
    "content":"bonjour",
    "mentions":[],
    "context":{
        "timezone":"",
        "email":"",
        "fullName":"",
        "username":"",
        "profilePictureUrl":""
    }
 }'
```
fail with the following error:
```
{"error":{"type":"invalid_request_error","message":"Invalid request body: Expecting NonEmptyString at 0.context.timezone but instead got: \"\",Expecting NonEmptyString at 0.context.username but instead got: \"\""}}
```
## Deploy Plan

Deploy `types`

### CODE REVIEW

File: types/src/front/api_handlers/public/assistant.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+import { NonEmptyString } from "io-ts-types";
 
 export const PublicPostMessagesRequestBodySchema = t.intersection([
   t.type({
     content: t.string,
     mentions: t.array(t.type({ configurationId: t.string })),
     context: t.type({
-      timezone: t.string,
-      username: t.string,
+      timezone: NonEmptyString,
```
Comment (spolu) [inline types/src/front/api_handlers/public/assistant.ts:9]: We also want to validate timezone manually to make sure it's a valid one on top of this non empty validation?

**Review by spolu — APPROVED** [review]
Looks good

We want to make sure we don't break the slack bot and the product :p

Do we agree that diff applied to both our endpoints? Public and internal?

**Comment by JulesBelveze** [comment]
@spolu what about the following:
```
const toNonEmptyString = (str: string): NonEmptyString | null => {
    const result = NonEmptyString.decode(str);
    return isRight(result) ? result.right : null;
  };
  const nonEmptyUsername = toNonEmptyString(user.username);

  if (!nonEmptyUsername) {
    return new Err(new Error("username must be a non-empty string"));
  }
  const convRes = await dustAPI.createConversation({
    title: transcriptTitle,
    visibility: "unlisted",
    message: {
      content: `:mention[${agent.name}]{sId=${agentConfigurationId}}`,
      mentions: [{ configurationId: agentConfigurationId }],
      context: {
        timezone: Intl.DateTimeFormat().resolvedOptions().timeZone ?? "UTC",
        username: nonEmptyUsername,
        fullName: user.name,
        email: user.email,
        profilePictureUrl: user.imageUrl,
      },
    },
```
here: https://github.com/dust-tt/dust/blob/fdf18bb0ed7c5e31c315810a965d348eaa59160d/front/temporal/labs/activities.ts#L248

**Comment by spolu** [comment]
It's fine to add code in the endpoint to validate vs solely relying on iots it easier of course

**Comment by JulesBelveze** [comment]
Alright, will merge it as it is tomorrow morning then 🦾

**Comment by spolu** [comment]
(If it complains on your machine it will likely complain at build time)

Not sure what you mean by merging "as is".

Did you test that:
- conversation creation still works in the product
- conversation creation by API still work with the timezone documented in the docs (I think it's "Europe/Paris")
- conversation creation still works with the slack bot (you can set it up in dev)

**Comment by spolu** [comment]
Also sorry for repeating myself (I believe): did we understand the initial failure / stream dropping error in the end?

**Comment by JulesBelveze** [comment]
I wasn't able to reproduce it. However my guess is that since the `timezone` could be `undefined` the error came from here https://github.com/dust-tt/dust/blob/fdf18bb0ed7c5e31c315810a965d348eaa59160d/front/lib/api/assistant/generation.ts#L512
Thus ensuring`timezone` to be defined would fix it. That's my best guess

**Comment by spolu** [comment]
> I wasn't able to reproduce it. However my guess is that since the `timezone` could be `undefined` the error came from here
> 
> https://github.com/dust-tt/dust/blob/fdf18bb0ed7c5e31c315810a965d348eaa59160d/front/lib/api/assistant/generation.ts#L512
> 
> 
> Thus ensuring`timezone` to be defined would fix it. That's my best guess

Why wouldn't be able to repro it then?

**Comment by JulesBelveze** [comment]
Don't know if my guess is wrong or if there's a possible state in which:
```
const d = moment(new Date()).tz(userMessage.context.timezone);
``` 
can be undefined (which I am not able to figure out) 🤷🏼‍♂️

**Comment by spolu** [comment]
Ok! Let's move forward with this and mergr once finalized and tested (as blast radius is quite large)

And we can consider the issue closed after that 👍

**Comment by JulesBelveze** [comment]
- conversation creation still works in the product ✅ 
- conversation creation by API still work with the timezone documented in the docs (I think it's "Europe/Paris") ✅ 
- conversation creation still works with the slack bot (you can set it up in dev)

**Comment by JulesBelveze** [comment]
@spolu Last review to make sure we aligned? 🙏🏼

**Comment by spolu** [comment]
As per IRL
- Labs diff likely unnecesasry
- Action: check that our users all have username
- Action: delurk the code
- Enforcement of non emptyness and timezone is desirable on the conversation endpoints of course
- Initial stream dropping errors are still NOT understood
- Action re-investigate them and reproduce them


cc @JulesBelveze and @lasryaric as eng runner

**Comment by JulesBelveze** [comment]
Created a fresh issue [here](https://github.com/orgs/dust-tt/projects/2/views/1?pane=issue&itemId=65989964)

**Pull request merged**